libs/state/src/lib/rx-state.service.ts

```diff
@@ -634,6 +684,62 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
     const subscription = new Subscription();
     subscription.add(this.accumulator.subscribe());
     subscription.add(this.effectObservable.subscribe());
+    this.createSignalStoreProxy();
     return subscription;
   }
+
+  private createSignalStoreProxy() {
+    const state$ = this.$;
+    const subs = this.subscription;
+    const currentState = (): T => this.get() || ({} as T);
+    const signalStore: SignalStore<T> = {} as SignalStore<T>;
+    this.signalStoreProxy = new Proxy<SignalStore<T>>(signalStore, {
+      get<K extends keyof T>(
+        target: SignalStore<T>,
+        p: K | string | symbol
+      ): Signal<T[K]> {
```

Comment:
Here we say `Signal<T[K]>` but in the `initialValue` we have `signal(val ?? null)`

Do we want to have it as `Signal<T[K] | null>` ?